contrib/sbtc/key-admin/contracts/key-admin.clar

```diff
@@ -0,0 +1,68 @@
+
+;; title: key-admin
+;; version:
+;; summary:
+;; description:
+
+;; traits
+;;
+;;(impl-trait tokens-ft)
+
+;; token definitions
+;; 
+(define-fungible-token tokens)
+
+;; constants
+;;
+(define-constant contract-owner tx-sender)
+(define-constant err-invalid-caller (err u1))
+
+;; data vars
+;;
+(define-data-var coordinator (optional principal) none)
+
+;; data maps
+;;
+(define-map signers uint principal)
+
+;; public functions
+;;
+(define-public (set-coordinator-key (key principal))
+    (if (is-valid-caller)
+        (ok (var-set coordinator (some key)))
+        err-invalid-caller
+    )
+)
+
+(define-public (set-signer-key (id uint) (key principal))
+    (if (is-valid-caller)
+        (ok (map-set signers id key))
+        err-invalid-caller
+    )
+)
+
+(define-public (mint! (amount uint))
+    (if (is-valid-caller)
+        (token-credit! tx-sender amount)
+        err-invalid-caller
+    )
+)
+
+;; read only functions
+;;
+(define-read-only (get-coordinator-key)
+  (var-get coordinator)
+)
+
+(define-read-only (get-signer-key (signer uint))
```

Comment:
Can we add a method to return the entire map?

Reply:
@igorsyl How do you return an entire map?  var-get doesn't work, and map-get expects a key.

Reply:
As discussed elsewhere, this is trivial as long as ```key_id```s are ```1..N```, otherwise it will be extremely difficult.  Since we are already using such ```key_id```s it seems logical to keep doing so.